### PR TITLE
XN-1532 add update messages parsing benchmarks

### PR DESCRIPTION
**Pull request by little-dude**
This adds benchmarks for the update messages. Here are some results from my laptop:

| seed dict entries | model size | total size | time (slice) | time (stream) |
| ---               | ---        | ---        | ---          | ---           |
| 1                 | 128        | ~272B      | 2.78 us      | 2.16 us       |
| 1000              | 6 kB       | ~1.1 kB    | 237 us       | 1.3 ms        |
| 10000             | 60 kB      | ~1.2 MB    | 2.3 ms       | 12.4 ms       |
| 10000             | ~1 MB      | ~2.1 MB    | 12.4 ms      | 30.1 ms       |
| 10000             | ~9 MB      | ~10.1 MB   | 98.7 ms      | 171.6 ms      |

Parsing from iterators is clearly slower (between 6x for small messages and 1.6x for large ones). I think the poor performance for small messages doesn't matter because we'll only parse multipart messages from iterators, so big ones a priori.

Based on these results, we won't be able to parse much more than a couple 10MB messages per second.

### CODE REVIEW

File: rust/benches/messages/utils.rs
Comment (janpetschexain) [inline rust/benches/messages/utils.rs:23]: this doesn't match the actual impl, should it be
```suggestion
    ///     - a 28 bytes serialized masked model
```
?

File: rust/benches/messages/update.rs
Comment (janpetschexain) [inline rust/benches/messages/update.rs:8]: it seems that all the `parse_...` functions could be generated by a macro. on the other hand it's just around 100 lines of code, so maybe not needed. up to you.

**Review by janpetschexain — APPROVED** [review]

**Comment by janpetschexain** [comment]
and the byte counts regarding the masks will change once #565 is merged, so we should decide on merge order such that the other can be rebased then.

File: rust/benches/messages/update.rs
Comment (little-dude) [reply rust/benches/messages/update.rs:8]: Agree it would be worth a macro, but I didn't want to put that much effort into it since it's just a benchmark.

File: rust/xaynet-core/src/testutils/multipart.rs
Comment (little-dude) [reply rust/xaynet-core/src/testutils/multipart.rs:36]: No particular reason for only using 8 bytes. This was just the easiest way I found to generate 2^64 unique keys (the previous version could only provide 255 unique keys).

File: rust/xaynet-core/src/testutils/multipart.rs
Comment (little-dude) [reply rust/xaynet-core/src/testutils/multipart.rs:38]: Exactly. I just want to convert `i` into an `u8`. I can change to a modulo if you think that makes the intent clearer.

**Comment by little-dude** [comment]
> and the byte counts regarding the masks will change once #565 is merged, so we should decide on merge order such that the other can be rebased then.

I'll let @finiteprods prod merge their PR first.

File: rust/xaynet-core/src/testutils/multipart.rs
Comment (janpetschexain) [reply rust/xaynet-core/src/testutils/multipart.rs:38]: there should be no need for explicit modulo as well, type casting `usize` values `i as u8` should already do this.

**Review by wilk10 — APPROVED** [review]
LGTM! thanks a lot

**Pull request merged**
